Engine/DataFeeds/SubscriptionDataReader.cs

```diff
@@ -448,16 +448,14 @@ private ISubscriptionDataSourceReader CreateSubscriptionFactory(SubscriptionData
 
         private void AttachEventHandlers(ISubscriptionDataSourceReader dataSourceReader, SubscriptionDataSource source)
         {
+            // NOTE: There seems to be some overlap in InvalidSource and CreateStreamReaderError
+            //       this may be worthy of further investigation and potential consolidation of events.
```

Comment:
Originally had the logging here but leaving this note to posterity. I suspect `InvalidSource` and `CreateStreamReaderError` may be able to be combined and properly abstracted vs needing to cast to a `TextSubscriptionDataSourceReader` in order to access the events.